### PR TITLE
[oc-147] - Add context to all git methods

### PR DESCRIPTION
**Pull request by ahrav**
- This will allow the recover to include information about the source that panics.
- Remove logrus and use custom logr.

### CODE REVIEW

File: pkg/sources/git/git.go [hunk 1]
```diff
@@ -338,11 +336,12 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 
 	var depth int64
 	var reachedBase = false
-	log.WithField("repo", urlMetadata).Debugf("Scanning repo")
+
+	ctx.Logger().V(1).Info("scanning repo", "repo", urlMetadata, "base", scanOptions.BaseHash, "head", scanOptions.HeadHash)
 	for commit := range commitChan {
-		log.Tracef("Scanning commit %s", commit.Hash)
+		ctx.Logger().V(2).Info("scanning commit", "commit", commit.Hash, "message", commit.Message)
```
Comment (mcastorina) [inline pkg/sources/git/git.go:342]: I would make this level 5 as it's very verbose.

File: pkg/sources/git/git.go [hunk 2]
```diff
@@ -443,9 +440,8 @@ func (s *Git) gitChunk(diff gitparse.Diff, fileName, email, hash, when, urlMetad
 			}
 		}
 
-		_, err := newChunkBuffer.Write(line)
-		if err != nil {
-			log.WithError(err).Error("Could not write line to git diff buffer.")
+		if _, err := newChunkBuffer.Write(line); err != nil {
+			ctx.Logger().V(1).Info("error writing to chunk buffer", "error", err, "filename", fileName, "commit", hash, "file", diff.PathB)
 		}
```
Comment (mcastorina) [inline pkg/sources/git/git.go:445]: Here we could use

```go
ctx.Logger().Error(err, "error writing to chunk buffer", "filename", fileName, ...)
```

which will be an error that gets sent to sentry. I think in this case it makes sense to do it how you have here though.

File: pkg/sources/git/git.go [hunk 3]
```diff
@@ -799,7 +801,7 @@ func getSafeRemoteURL(repo *git.Repository, preferred string) string {
 }
 
 func handleBinary(ctx context.Context, repo *git.Repository, chunksChan chan *sources.Chunk, chunkSkel *sources.Chunk, commitHash plumbing.Hash, path string) error {
-	log.WithField("path", path).Trace("Binary file found in repository.")
+	ctx.Logger().V(1).Info("handling binary file", "path", path)
```
Comment (mcastorina) [inline pkg/sources/git/git.go:804]: Maybe this should be level 5 too? Not sure how often it will log this.

File: pkg/sources/git/git.go (hunk 2 above)
Comment (ahrav) [reply pkg/sources/git/git.go:445]: Ahh good to know, I was unaware of that actually.

**Review by dustin-decker — APPROVED** [review]

**Pull request merged**